### DISCUSSION TITLE
different gene names/symbols for each isolate of same bacterial species after simulating single gene knockout

### DISCUSSION THREAD

**Discussion by Zhaoju-Deng** [opencobra/cobrapy#1389] [Q&A] [answered]
dear all, 
I am quite new in genome scale metabolic modeling, I constructed GEMs from multiple reference genomes of  _Strep. agalactiae_  using Carveme using the default settings and simulated single gene/reaction knockout using cobrapy for all xml files constructed from whole genome sequences, I am trying to  pool the knockout results, while the gene names/symbols are different across isolates, are there any tools to specifically unify the gene/reaction names/symbols for all isolates?
 many thanks,
Zhaoju

**Comment by cdiener** [comment, marked as answer]
Hi, since CARVME maps to a unified universal model there should be annotations in the model that identify the genes. You could extract those.

Example:

```python
In [9]: from cobra.io import read_sbml_model

In [10]: mod = read_sbml_model("~/Downloads/SRR9205533.asm1.xml.gz")

In [11]: mod.genes[0]
Out[11]: <Gene k141_54_256 at 0x7fe1f488f1d0>

In [12]: mod.genes[0].annotation
Out[12]: 
{'sbo': 'SBO:0000243',
 'interpro': ['IPR001670',
  'IPR012079',
  'IPR015590',
  'IPR016161',
  'IPR016162',
  'IPR016163',
  'IPR018211'],
 'uniprot': ['Q184F6', 'Q184J7', 'P0A9Q7', 'P0A9S1', 'P37686'],
 'goa': ['Q184F6', 'Q184J7', 'P71017'],
 'asap': ['ABE-0004164', 'ABE-0009177', 'ABE-0011725'],
 'ncbigene': ['945837', '947273', '948102', '852636'],
 'ecogene': ['EG10031', 'EG10351', 'EG12293'],
 'sgd': 'S000003225',
 'subtilist': 'BG11941'}
```

**Comment by Zhaoju-Deng** [comment]
Hi Diener,
really appreciate for all your help! this really saved me lot of time!
